types/mempool/sender_nonce.go

```diff
@@ -65,6 +87,12 @@ func (snm *senderNonceMempool) setSeed(seed int64) {
 // Insert adds a tx to the mempool. It returns an error if the tx does not have at least one signer.
 // priority is ignored.
 func (snm *senderNonceMempool) Insert(_ sdk.Context, tx sdk.Tx) error {
+	if snm.maxTx > 0 && len(snm.existingTx) >= snm.maxTx {
+		return ErrMempoolTxMaxCapacity
+	}
+	if snm.maxTx < 0 {
+		return nil
+	}
```

Comment:
@odeke-em, what do you think of adding this extra condition, the result being, negative means unbounded, positive bounded, and 0 disabled? Given that the main request was the ability to turn it off. also what do you think @tac0turtle